src/main/kotlin/no/capraconsulting/siren/Action.kt

```diff
@@ -0,0 +1,231 @@
+import no.capraconsulting.siren.internal.util.skipNulls
+
+/**
+ * Actions show available behaviors an entity exposes and are used for executing
+ * state transitions. Represented in JSON Siren as an array such as
+ * `{ "actions": [{ ... }] }`.
+ *
+ * **See also:** [Action specification](https://github.com/kevinswiber/siren.actions-1)
+ *
+ * Prefer using [newBuilder] to create a new instance instead of using the
+ * constructor, as the constructor has some relaxed checking which should
+ * only be used when deserializing an existing representation.
+ */
+data class Action(
+    /**
+     * A string that identifies the action to be performed. Action names MUST be
+     * unique within the set of actions for an entity. The behaviour of clients
+     * when parsing a Siren document that violates this constraint is undefined.
+     * Required.
+     *
+     * @return the value of name attribute
+     */
+    val name: String,
+    /**
+     * Describes the nature of an action based on the current representation.
+     * Possible values are implementation-dependent and should be documented.
+     *
+     * @return the value of class attribute or an empty list if it is missing
+     */
+    val clazz: List<String> = emptyList(),
+    /**
+     * An enumerated attribute mapping to a protocol method. For HTTP, these
+     * values may be GET, PUT, POST, DELETE, or PATCH. As new methods are
+     * introduced, this list can be extended. If this attribute is
+     * omitted, GET should be assumed.
+     *
+     * @return the value of method attribute
+     */
+    val method: String? = null,
+    /**
+     * The URI of the action. Required.
+     *
+     * @return the value of href attribute
+     */
+    val href: URI,
+    /**
+     * Descriptive text about the action.
+     *
+     * @return the value of title attribute
+     */
+    val title: String? = null,
+    /**
+     * The encoding type for the request. When omitted and the fields attribute
+     * exists, the default value is `application/x-www-form-urlencoded`.
+     *
+     * @return the value of type attribute
+     */
+    val type: String? = null,
+    /**
+     * A collection of fields.
+     *
+     * @return the value of fields attribute or an empty list if it is missing
+     */
+    val fields: List<Field> = emptyList()
+) : Serializable {
+
+    internal fun toRaw(): Map<String, Any?> =
+        LinkedHashMap<String, Any?>().apply {
+            this[Siren.NAME] = name
+            this[Siren.TITLE] = title
+            this[Siren.CLASS] = if (clazz.isEmpty()) null else clazz
+            this[Siren.METHOD] = method
+            this[Siren.HREF] = href
+            this[Siren.TYPE] = type
+            this[Siren.FIELDS] = if (fields.isEmpty()) null else fields.map(Field::toRaw)
+        }.skipNulls()
+
+    /**
+     * Builder for [Action].
+     */
+    class Builder internal constructor(private val name: String, private val href: URI) {
```

Comment:
Builder-pattern er vel overflødig når man har defaultverdier?

Reply:
Ja, i Kotlin. Men biblioteket her er skrevet for å bli brukt fra Java, hvor man ikke kan bruke default parameters.

(I tillegg så opprettholder denne Kotlin-migrasjonen kompabilitet mot Java-versjonen som har dette. Har unngått breaking changes.)